app/rest/controllers/src/main/java/io/syndesis/controllers/logs/LogsController.java

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.controllers.logs;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodList;
+import io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl;
+import io.fabric8.kubernetes.client.utils.HttpClientUtils;
+import io.fabric8.openshift.client.NamespacedOpenShiftClient;
+import io.syndesis.core.Json;
+import io.syndesis.core.KeyGenerator;
+import io.syndesis.jsondb.GetOptions;
+import io.syndesis.jsondb.JsonDB;
+import io.syndesis.openshift.OpenShiftService;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.skife.jdbi.v2.DBI;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This class tracks pod logs and ingests them into our DB.
+ */
+@Service
+@ConditionalOnProperty(value = "controllers.logs.enabled", havingValue = "true", matchIfMissing = false)
+public class LogsController implements Closeable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LogsController.class);
+
+    private final DBI dbi;
+    private final NamespacedOpenShiftClient client;
```

Comment:
We could live here with just an OpenshiftClient here (as in 99% of the cases).

The NamespacedClient variation only provides the ability to obtain a new instance of the client that defaults to an other namespace and I don't think anyone is really using that.

Reply:
Might as well go down to KubernetesClient actually.